libwaysip/src/lib.rs

```diff
@@ -36,19 +36,12 @@ fn get_cursor_buffer(connection: &Connection, shm: &WlShm) -> Option<CursorImage
 }
 
 /// get the selected area
-pub fn get_area(kind: SelectionType) -> Result<Option<state::AreaInfo>, WaySipError> {
-    let connection =
-        Connection::connect_to_env().map_err(|e| WaySipError::InitFailed(e.to_string()))?;
-    let (globals, _) = registry_queue_init::<state::WaysipState>(&connection)
-        .map_err(|e| WaySipError::InitFailed(e.to_string()))?; // We just need the
-                                                               // global, the
-                                                               // event_queue is
-                                                               // not needed, we
-                                                               // do not need
-                                                               // state::BaseState after
-                                                               // this anymore
-
-    let mut state = state::WaysipState::new(kind);
+pub fn get_area(
```

Comment:
I think the origin api should be reserved.Sometimes  I do not want to induce wayland-client

Reply:
maybe add like `get_area_with_connection` like this kind , maybe add one extra function will be better I think.

Reply:
or make connection and globals optional I think will be better. if it is none, make a new connection and globals

like create an enum

```rust
WaysipWaylandInfo<'a> {
    globals: GlobalList,
   connection: &'a Connection
}
```